modules/apps/frontend-taglib/frontend-taglib/src/test/java/com/liferay/frontend/taglib/servlet/taglib/ComponentTagTest.java

```diff
@@ -0,0 +1,161 @@
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.servlet.taglib;
+
+import com.liferay.frontend.js.loader.modules.extender.npm.NPMResolvedPackageNameUtil;
+import com.liferay.portal.json.JSONFactoryImpl;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.portlet.LiferayPortletConfig;
+import com.liferay.portal.kernel.servlet.taglib.aui.ScriptData;
+import com.liferay.portal.kernel.theme.PortletDisplay;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.JavaConstants;
+import com.liferay.portal.kernel.util.PortalClassLoaderUtil;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.kernel.util.PropsUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.util.PortalImpl;
+import com.liferay.portal.util.PropsImpl;
+
+import java.io.StringWriter;
+
+import java.net.URL;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import java.util.Collections;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.mockito.Mockito;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockPageContext;
+
+/**
+ * @author Cristina González
+ */
+public class ComponentTagTest {
+
+	@BeforeClass
+	public static void setUpClass() {
+		JSONFactoryUtil jsonFactoryUtil = new JSONFactoryUtil();
+
+		jsonFactoryUtil.setJSONFactory(new JSONFactoryImpl());
+
+		PortalClassLoaderUtil.setClassLoader(PortalImpl.class.getClassLoader());
+
+		PortalUtil portalUtil = new PortalUtil();
+
+		portalUtil.setPortal(new PortalImpl());
+
+		PropsUtil.setProps(new PropsImpl());
+	}
+
+	@Test
+	public void testDoEndTag() throws Exception {
+		ComponentTag componentTag = new ComponentTag();
+
+		HttpServletRequest httpServletRequest = _getHttpServletRequest();
+
+		componentTag.setComponentId("componentId");
+		componentTag.setContext(Collections.singletonMap("name", "value"));
+		componentTag.setModule("module");
+		componentTag.setPageContext(
+			new MockPageContext(_getServletContext(), httpServletRequest));
+
+		componentTag.doEndTag();
+
+		Document document = _toDocument(
+			(ScriptData)httpServletRequest.getAttribute(
+				WebKeys.AUI_SCRIPT_DATA));
+
+		Elements elements = document.select("script[type='text/javascript']");
+
+		Assert.assertEquals(1, elements.size());
+
+		Element element = elements.get(0);
+
+		Assert.assertEquals(_expectedResult(), element.html());
```

Comment:
I would probably relax this assertion a bit, since the `Liferay.Loader.require` and `Liferay.Loader.component` scaffolding can change for reasons outside of the scope of the text. However given that it is a single test and that the aforementioned changes are unlikely, we may go well with it like this.

Reply:
Thanks a lot for the review :). I have found this point really interesting, do you have any idea of how could I relax the condition?, do you know any lib to parse JS from Java?

Thanks!

Reply:
:thinking: I was thinking of using substrings, not really parsing the JS code. 

Seeing the test again..., I assumed yesterday that you just wanted to test that the map is passed, thus I was thinking in just looking for:

```
module.default({"name":"value","namespace":"namespace","spritemap":"\/lexicon\/icons.svg"})
```

But now I see you called your test `testDoEndTag` so I guess it is correct to check for everything :-). I don't think it is worth the effort to obtain a JS AST just to make sure the test doesn't fail because of whitespace since it would make the test more complex and still it would fail if something significant is changed.

Reply:
OK, thanks a lot for the review :)